File: 프로그래머스/lv2/12946. 하노이의 탑/하노이의 탑.py
def dfs(n, start, end, mid):
    global result
    
    if n == 1:
        return result.append([start, end])
    dfs(n-1, start, mid, end)
    result.append([start, end])
    dfs(n-1, mid, end, start)


def solution(n):
    global result
    result = []
    dfs(n, 1, 3, 2)
    return result

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.java
import java.util.*;


class Solution {
    List<int[]> list = new ArrayList<>();
    public int[][] solution(int n) {

        dfs(n, 1,2,3);

        int[][] answer = list.stream().toArray(int[][]::new);
        return answer;
    }

    private void dfs(int n, int start, int mid, int end) {
        if(n==0){
            return;
        }

        dfs(n-1, start, end, mid);
        list.add(new int[]{start,end});
        dfs(n -1, mid, start,end);
    }
}